src/main/java/dataStructures/hashSet/openAddressing/HashSet.java

```diff
@@ -121,15 +134,29 @@ public boolean remove(T element) {
      * @return true if this set contains the specified element
      */
     public boolean contains(T element) {
-        int bucketIndex = this.search(element);
+        int collisions = 0;
+        while (collisions < capacity()) {
+            int bucketIndex = hashFunction(element, collisions);
+
+            // Invariant: Probe sequence is unbroken (no null values between buckets in the sequence).
```

Comment:
for long comments to explain any concerns arising due to our implementation, can put in method docs? see the sorting functions. Rationale for doing so is so that the portion within the braces doesnt get unnecessarily extended. But pls share if u find any other concerns.